HW2/P2/P2.py

```diff
@@ -69,14 +77,41 @@
         move_data_fine_grained(counts, src, dest, 100)
     assert counts.sum() == total, "Wrong total after move_data_fine_grained"
     print("Fine grained correlated: {} seconds".format(t.interval))
+    fine_grained_correlated = t.interval
+
 
     ########################################
     # You should explore different values for the number of locks in the medium
     # grained locking
     ########################################
-    N = 10
-    counts[:] = orig_counts
-    with Timer() as t:
-        move_data_medium_grained(counts, src, dest, 100, N)
-    assert counts.sum() == total, "Wrong total after move_data_medium_grained"
-    print("Medium grained correlated: {} seconds".format(t.interval))
+    coarse_correlated = {}
+    for N in N_values:
+        counts[:] = orig_counts
+        with Timer() as t:
+            move_data_medium_grained(counts, src, dest, 100, N)
+        assert counts.sum() == total, "Wrong total after move_data_medium_grained"
+        print("Medium grained correlated (N={}): {} seconds".format(N, t.interval))
+        coarse_correlated[N] = t.interval
+
+    # display results
+    keys, values = zip(*sorted(coarse_correlated.items(), key = lambda x: x[0]))
+    values = [serial_correlated, fine_grained_correlated] + list(values)
+    labels = ['Serial', 'Fine Grained'] + ['N=%d' % n for n in keys]
+    positions = np.arange(len(values))
+    plt.bar(positions, values, align='center')
+    plt.xticks(positions, labels)
+    plt.ylabel('Time to move data')
+    plt.title('Correlated Data Movement Times')
+    plt.show()
+
+    keys, values = zip(*sorted(coarse_uncorrelated.items(), key = lambda x: x[0]))
+    values = [serial_uncorrelated, fine_grained_uncorrelated] + list(values)
+    labels = ['Serial', 'Fine Grained'] + ['N=%d' % n for n in keys]
+    positions = np.arange(len(values))
+    plt.bar(positions, values, align='center')
+    plt.xticks(positions, labels)
+    plt.ylabel('Time to move data')
+    plt.title('Uncorrelated Data Movement Times')
```

Comment:
Might just be easier to put the uncorrelated and correlated times on the same graph with 2 line graphs.